Automattic/newspack-newsletters
# feat: email verification and subscription management

### All Submissions:






### Changes proposed in this Pull Request:

Implements email verification for subscription management through WooCommerce's "My Account" page.

For reviewing purposes, this PR implements subscription management only for ActiveCampaign at the moment. Other ESPs should follow.

<img width="1097" alt="image" src="https://user-images.githubusercontent.com/820752/179840385-b0491c8d-7757-46c2-8c0b-97d6e3cab657.png">

<img width="1228" alt="image" src="https://user-images.githubusercontent.com/820752/179840423-403fef79-5555-4dd2-a648-a81919446a2a.png">

<img width="1315" alt="image" src="https://user-images.githubusercontent.com/820752/179840528-ea43f68d-85b9-49f3-ac02-497c314affe8.png">

<img width="1109" alt="image" src="https://user-images.githubusercontent.com/820752/179841680-d70b8109-cbad-4bcd-9f1b-68ed2b589d8b.png">



Partial of https://github.com/Automattic/newspack-plugin/issues/1743

### How to test the changes in this Pull Request:

1. Make sure you have WC installed and active, a way to intercept WP's sent emails, and ActiveCampaign as your ESP
2. Visit the "Engagement" wizard and make sure you have at least one subscription list enabled
3. Visit "My Account" page and click on "Newsletters"
4. Confirm you see a button to confirm your email
5. Click on the button and confirm you receive an email with a link
6. Click on the link and confirm the success message and the list of subscription lists to toggle
7. Toggle the lists (unsubscribe to one and subscribe to another)
8. Confirm the changes are reflected on the page and on the ESP dashboard

### Changing email

1. Change your account's email
2. Visit the Newsletters page again and confirm you have to verify again
3. Verify the email and confirm you can manage the subscriptions

### Unsupported ESP

1. Change your ESP to Mailchimp
4. Visit the "My Account" page and confirm you **cannot** see the "Newsletter" link

### Other information:
